src/loops.rs

```diff
@@ -0,0 +1,89 @@
+use super::Monad;
+
+pub fn bind_for_loop<'m, C, M, NextM>(
+	container: C,
+	body: impl FnMut(C::Item) -> M + 'm,
+) -> NextM
+where
+	C: IntoIterator,
+	C::IntoIter: Clone + 'm,
+	M: Monad<'m, LoopControl, Bind<LoopControl> = NextM>,
+	NextM: Monad<'m, LoopControl>,
+{
+	let iterator = container.into_iter();
+	recurse_for_loop(iterator, body)
```

Comment:
This returns `M LoopControl` , change this to `M ()` so LoopControl does not leak out

Reply:
Done!